### PR TITLE
Progress Tab:Show "Update required" for app version less than '2022-07-04-8318'

### PR DESCRIPTION
**Pull request by PriyangaPKini**
**Story card:** [sc-9741](https://app.shortcut.com/simpledotorg/story/9741/show-old-progress-tab-when-app-version-2022-07-04-8318)

## Because

APP versions < '2022-07-04-8318' crashes when rendering the new progress tab.

## This addresses

Show "Updated required" for APP version < '2022-07-04-8318' when the new progress tab is enabled. 
In Srilanka and Ethiopia, we will still show the old progress tab.

#### Setup
- Use postman/mod header to pass additional headers. You will need to pass these headers below
  - `X-USER-ID`,  `X-Facility-ID`, `Authorization`. Use `rails get_user_crendentials` to get the values for these headers locally
- Goto `http://localhost:3000/api/v3/analytics/user_analytics.html#`
- Add the header `HTTP-X-APP-VERSION`:  `2022-06-04-8318`. You should see the old progress tab
- Add the header `HTTP-X-APP-VERSION`:  `2022-07-04-8318`. You should see the new progress tab

## Screenshots

**When `new_progress_tab` is enabled and app version >= '2022-07-04-8318'**

Screenshot 2022-12-09 at 1 37 41 PM

**When `new_progress_tab` is enabled and app version <  '2022-07-04-8318'**

Screenshot 2022-12-09 at 1 37 31 PM

**When `new_progress_tab` is disabled (For Ethiopia and Srilanka)**

Screenshot 2022-12-09 at 1 37 07 PM

### CODE REVIEW

**Review by harimohanraj89 — CHANGES_REQUESTED** [review]
Some feedbacks

File: app/controllers/api/v3/analytics/user_analytics_controller.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@ class Api::V3::Analytics::UserAnalyticsController < Api::V3::AnalyticsController
   before_action :set_for_end_of_month
   before_action :set_bust_cache
 
+  APP_OLD_VERSION_RELEASE_DATE = "2022-07-04"
```
[the remark this answers is not in the document]
Comment (PriyangaPKini) [reply app/controllers/api/v3/analytics/user_analytics_controller.rb:7]: Hmm, that make sense.

File: app/controllers/api/v3/analytics/user_analytics_controller.rb [hunk 2]
```diff
@@ -55,4 +55,10 @@ def show
   def set_bust_cache
     RequestStore.store[:bust_cache] = true if params[:bust_cache].present?
   end
+
+  def new_progress_tab_enabled?
+    app_version = request.headers["HTTP-X-APP-VERSION"] || "" # Default to old progress tab when app version is nil
+    (app_version >= APP_OLD_VERSION_RELEASE_DATE) &&
+      (Flipper.enabled?(:new_progress_tab_v2, current_user) || Flipper.enabled?(:new_progress_tab_v2))
```
[the remark this answers is not in the document]
Comment (PriyangaPKini) [reply app/controllers/api/v3/analytics/user_analytics_controller.rb:62]: Oh no. Will fix this

File: app/controllers/api/v3/analytics/user_analytics_controller.rb [hunk 3]
```diff
@@ -55,4 +55,10 @@ def show
   def set_bust_cache
     RequestStore.store[:bust_cache] = true if params[:bust_cache].present?
   end
+
+  def new_progress_tab_enabled?
+    app_version = request.headers["HTTP-X-APP-VERSION"] || "" # Default to old progress tab when app version is nil
```
[the remark this answers is not in the document]
Comment (PriyangaPKini) [reply app/controllers/api/v3/analytics/user_analytics_controller.rb:60]: That's a good point. I'll do this way

File: app/controllers/api/v3/analytics/user_analytics_controller.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (harimohanraj89) [reply app/controllers/api/v3/analytics/user_analytics_controller.rb:62]: No worries! It was still working as intended before 👍

[the remark this answers is not in the document]
Comment (PriyangaPKini) [reply app/controllers/api/v3/analytics/user_analytics_controller.rb:62]: I have done this mistake in more places, I will fix that too in a separate PR

File: app/controllers/api/v3/analytics/user_analytics_controller.rb [hunk 4]
```diff
@@ -37,10 +39,8 @@ def show
     @diabetes_reports_data = Reports::ReportsFakeFacilityProgressService.new(@current_facility.name).diabetes_reports
 
     respond_to do |format|
-      if Flipper.enabled?(:new_progress_tab_v2, current_user) || Flipper.enabled?(:new_progress_tab_v2)
+      if new_progress_tab_enabled?
         format.html { render :show_v2 }
-      elsif Flipper.enabled?(:new_progress_tab_v1, current_user) || Flipper.enabled?(:new_progress_tab_v1)
-        format.html { render :show_v1 }
       else
         format.html { render :show }
```
Comment (PriyangaPKini) [inline app/controllers/api/v3/analytics/user_analytics_controller.rb:45]: Show "Update required" instead of old progress tab

**Comment by jamiecarter7** [comment]
Implemented 'update app' card:
- card design
- translation strings
please check I've done this all correctly as I am unable to test on localhost now the app version thing has been implemented

File: app/views/api/v3/analytics/user_analytics/progress_update_required.html.erb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  <%= inline_stylesheet("user_analytics.css") %>
+</head>
+<body class="bgc-grey-lightest">
+<div class="mb-8px p-16px pt-48px pb-24px bgc-white br-4px bs-card">
+  <%= inline_svg("warning_sign.svg") %>
+  <p class="fs-20px fw-bold mt-32px mb-12px"><%= t("progress_tab.update_required_card.title") %></p>
+  <p class="fs-16px c-grey-dark"><%= t("progress_tab.update_required_card.subtitle") %></p>
+</div>
+</body>
+<%= inline_js("standalone/user_analytics.js") %>
```
Comment (prabhanshuguptagit) [inline app/views/api/v3/analytics/user_analytics/progress_update_required.html.erb:23]: Think this view doesn't need to include the JS.

**Review by prabhanshuguptagit — APPROVED** [review]

File: app/controllers/api/v3/analytics/user_analytics_controller.rb (hunk 4 above)
Comment (PriyangaPKini) [reply app/controllers/api/v3/analytics/user_analytics_controller.rb:49]: Show the **_old progress tab_** when `new_progress_tab` is disabled (In Ethiopia and Srilanka)
Show **_"Update Required"_** when `new_progress_tab` is enabled and `app version < min_app_version`
Show the **_new progress tab_** when `new_progress_tab` is enabled and `app version >= min_app_version`

**Pull request merged**